core/bin/snapshots_creator/src/main.rs

```diff
@@ -0,0 +1,273 @@
+
+    snapshot_l1_batch: Gauge<u64>,
+
+    #[metrics(buckets = Buckets::LATENCIES)]
+    storage_logs_processing_durations: Histogram<Duration>,
+
+    #[metrics(buckets = Buckets::LATENCIES)]
+    factory_deps_processing_durations: Histogram<Duration>,
+}
+#[vise::register]
+pub(crate) static METRICS: vise::Global<SnapshotsCreatorMetrics> = vise::Global::new();
+
+async fn maybe_enable_prometheus_metrics(stop_receiver: Receiver<bool>) -> anyhow::Result<()> {
+    let prometheus_config = PrometheusConfig::from_env().ok();
+    if let Some(prometheus_config) = prometheus_config {
+        let exporter_config = PrometheusExporterConfig::push(
+            prometheus_config.gateway_endpoint(),
+            prometheus_config.push_interval(),
+        );
+
+        tracing::info!("Starting prometheus exporter with config {prometheus_config:?}");
+        tokio::spawn(exporter_config.run(stop_receiver));
+    } else {
+        tracing::info!("Starting without prometheus exporter");
+    }
+    Ok(())
+}
+async fn process_storage_logs_single_chunk(
+    blob_store: &dyn ObjectStore,
+    pool: &ConnectionPool,
+    miniblock_number: MiniblockNumber,
+    l1_batch_number: L1BatchNumber,
+    chunk_id: u64,
+    chunk_size: u64,
+    chunks_count: u64,
+) -> anyhow::Result<String> {
+    let latency = METRICS.storage_logs_processing_durations.start();
+    let mut conn = pool.access_storage_tagged("snapshots_creator").await?;
+    let logs = conn
+        .snapshots_creator_dal()
+        .get_storage_logs_chunk(miniblock_number, chunk_id, chunk_size * chunk_id)
+        .await
+        .context("Error fetching storage logs count")?;
+    let storage_logs_chunk = SnapshotStorageLogsChunk { storage_logs: logs };
+    let key = SnapshotStorageLogsStorageKey {
+        l1_batch_number,
+        chunk_id,
+    };
+    let filename = blob_store
+        .put(key, &storage_logs_chunk)
+        .await
+        .context("Error storing storage logs chunk in blob store")?;
+
+    let output_filepath_prefix = blob_store.get_storage_prefix::<SnapshotStorageLogsChunk>();
+    let output_filepath = format!("{output_filepath_prefix}/{filename}");
+
+    tracing::info!(
+        "Finished storing storage logs chunk {}/{chunks_count}, output stored in {output_filepath}",
+        chunk_id + 1,
+    );
+    latency.observe();
+    Ok(output_filepath)
+}
+
+async fn process_factory_deps(
+    blob_store: &dyn ObjectStore,
+    pool: &ConnectionPool,
+    miniblock_number: MiniblockNumber,
+    l1_batch_number: L1BatchNumber,
+) -> anyhow::Result<String> {
+    let latency = METRICS.factory_deps_processing_durations.start();
+    tracing::info!("Processing factory dependencies");
+    let mut conn = pool.access_storage_tagged("snapshots_creator").await?;
+    let factory_deps = conn
+        .snapshots_creator_dal()
+        .get_all_factory_deps(miniblock_number)
+        .await?;
+    let factory_deps = SnapshotFactoryDependencies { factory_deps };
```

Comment:
You can drop a connection here so that it's not retained when not required (over persisting a potentially bulky file in GCS). Ditto for `process_storage_logs_single_chunk`.

Reply:
corrected